src/mscorlib/shared/System/IO/StreamWriter.cs

```diff
@@ -325,10 +328,7 @@ public override void Write(char value)
 
         public override void Write(char[] buffer)
```

Comment:
Is the inliner smart enough to not inline WriteSpan when it inlines (de-virtualized) Write?

Reply:
It appears it inlines WriteSpan into Write and then doesn't inline Write into its caller, as desired.  Do you want me to add NoInlining to all of the call sites of WriteSpan?
cc: @AndyAyersMS

Reply:
I would expect if `Write` gets devirtualized and inlined then `WriteSpan` would also get inlined, as `AggressiveInlining` doesn't get cancelled out. But it's probably not super-common for `Write` to be devirtualized as we have to know we have exactly a `StreamReader` (say, the constructor call needs to be in the same method).

Adding `NoInlining` to the callers of `WriteSpan` is a reasonable way to limit the amount of code that gets duplicated.

Reply:
Thanks. Added.